admin-dev/themes/new-theme/js/product-page/combination.js

```diff
@@ -0,0 +1,123 @@
+import $ from 'jquery';
+
+export default function() {
+  $(document).ready(function() {
+    let $jsCombinationsList = $('.js-combinations-list');
+    let idsProductAttribute = $jsCombinationsList.data('ids-product-attribute').toString().split(',');
+    let idsCount = idsProductAttribute.length;
+    let currentCount = 0;
+    let step = 5;
+
+    $.get($jsCombinationsList.attr('data-action-refresh-images') + '/' + $jsCombinationsList.data('id-product'))
+      .then(function(response) {
+        getCombinations(response);
+        $('#create-combinations').click(function() {
+          generate();
+        });
+      });
+
+    let combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
+
+    let getCombinations = (combinationsImages) => {
+      $.get(combinationUrl).then(function (resp) {
+        $('#loading-attribute').before(resp);
+        refreshImagesCombination(combinationsImages, idsProductAttribute.slice(currentCount, currentCount+step));
+        currentCount += step;
+        combinationUrl = $jsCombinationsList.data('combinations-url') + '/' + idsProductAttribute.slice(currentCount, currentCount+step).join('-');
```

Comment:
I don't think it's really useful to use a variable because the `currentCount` is changing at every call. I've added a comment above explaining what is `idsProductAttribute.slice(currentCount, currentCount+step).join('-')`. This is the set of combinations to load